adapter/thing_management.md

```diff
@@ -0,0 +1,404 @@
+# Thing Management
+
+The following document describes the interaction flows and FIMP specification used in management of things.
+
+## Flows
+
+### Network Inclusion
+
+Network inclusion flow is used by wireless network adapters, such as Z-Wave and Zigbee, to include new devices into the network and add them to the system.
+
+* Client sends `cmd.thing.inclusion` command to the adapter with `true` value to start inclusion process and switch adapter into pairing mode.
+* User interacts with a physical device to switch it into pairing mode, e.g. clicks a button on a device three times in a quick sequence.
+* At any time the adapter can generate `evt.thing.inclusion_status_report` event to report the current status of the inclusion process.
+* If pairing of the device requires PIN and client support PIN request, the adapter generates `evt.thing.pin_needed` report.
+    * Client sends `cmd.thing.pin` command with the PIN to the adapter.
+* Once inclusion and interview of the device is finished adapter generates `evt.thing.inclusion_report` event with the thing specification.
+* Client sends `cmd.thing.inclusion` command to the adapter with `false` value to stop the inclusion process. Regardless, adapter should cancel inclusion after a specified timeout.
+
+### Smart Start Inclusion
+
+Smart Start inclusion is a Z-Wave specific flow allowing to preliminarily add a device to a provisioning list so that it can be automatically included once the device is turned on.
+See [Smart Start](/adapter/zwave_smart_start.md) documentation for more details.
+
+### Custom Inclusion
+
+Adapters utilizing custom protocol communication through local network or third-party cloud will have their own specific inclusion flow specified it in their own documentation.
+Preliminary steps of such flows will often include authentication with a third-party system, specific configuration, and selecting devices to be connected to the system.
+At the end of this process the adapter is always required to generate `evt.thing.inclusion_report` event with the thing specification.
+
+### Network Exclusion
+
+Network exclusion flow is used by wireless network adapters to exclude devices from the network and remove them from the system.
+
+* Client sends `cmd.thing.exclusion` command to the adapter to start exclusion process and switch adapter into unpairing mode.
+* User interacts with a physical device to switch it into unpairing mode, e.g. clicks a button on a device three times in a quick sequence.
+* At any time the adapter can generate `evt.thing.exclusion_status_report` event to report the current status of the exclusion process.
+* Once exclusion is finished adapter generates `evt.thing.exclusion_report` event with the excluded thing address.
+
+### Forced Exclusion
+
+If a device is offline and unreachable or the adapter does not support the network exclusion flow, the device can be force deleted from the system using this flow.
```

Comment:
Description should be more general imo. Overall description should say that some adapters might implement force exclusion of devices.
One of dots below this comment should describe this process from Z-Wave perspective, like:
- Z-Wave supports force remove of failing devices, which means if device is considered to be offline, then the removal will be proceeded with success.